### PR TITLE
Update docs for Ruby and Rust support

### PR DESCRIPTION
**Pull request by macraig**
## Test plan
Docs

### CODE REVIEW

File: doc/code_navigation/explanations/auto_indexing.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ With Sourcegraph deployments supporting [executors](../../admin/executors.md), y
 
 ## Language support
 
-Auto-indexing is currently available for Go, TypeScript, JavaScript and Python repositories. See also [dependency navigation](features.md#dependency-navigation) for instructions on how to setup cross-dependency navigation depending on what language ecosystem you use.
+Auto-indexing is currently available for Go, TypeScript, JavaScript, Python and Ruby repositories. See also [dependency navigation](features.md#dependency-navigation) for instructions on how to setup cross-dependency navigation depending on what language ecosystem you use.
```
Comment (varungandhi-src) [inline doc/code_navigation/explanations/auto_indexing.md:26]: We have not implemented auto-inference for Ruby yet, so this isn't quite correct (but it should be easy to add). The JSON config required a manual configuration step which is what auto-inference handles.

Comment (macraig) [reply doc/code_navigation/explanations/auto_indexing.md:26]: Leaving as is after live sync

Comment (macraig) [inline doc/code_navigation/explanations/auto_indexing.md:26]:
```suggestion
Auto-indexing is currently available for Go, TypeScript, JavaScript, Python, Ruby and JVM repositories. See also [dependency navigation](features.md#dependency-navigation) for instructions on how to setup cross-dependency navigation depending on what language ecosystem you use.
```

File: doc/code_navigation/index.md [hunk 2]
```diff
@@ -75,16 +75,19 @@ Here's how you go from search-based code navigation to **automatically-updating,
...
+    - [Index a C++ repository](how-to/index_a_cpp_repository.md)
+
 
 1. _Optional_: automate the uploading of indexes by [adding indexing and uploading to your CI setup](how-to/adding_lsif_to_workflows.md).
 1. Get **automatic precise code navigation** by [enabling auto-indexing](how-to/enable_auto_indexing.md) which makes Sourcegraph automatically index the your repositories and enable precise code navigation for them.
 1. Setup **auto-dependency indexing** to navigate and search through the dependencies used by your code:
     - **Go**: Enable [auto-indexing](explanations/auto_indexing.md) and Sourcegraph will start indexing your dependencies.
-    - **Java, Scala, Kotlin**: Enable [auto-indexing](explanations/auto_indexing.md) and setup a [JVM dependencies code host](../../integration/jvm.md).
     - **JavaScript, TypeScript**: Enable [auto-indexing](explanations/auto_indexing.md) and setup a [npm dependencies code host](../../integration/npm.md).
+    - **Java, Scala, Kotlin**: Enable [auto-indexing](explanations/auto_indexing.md) and setup a [JVM dependencies code host](../../integration/jvm.md).
```
Comment (macraig) [inline doc/code_navigation/index.md:89]: @varungandhi-src should we add a link to the steps to add a Ruby config?

Comment (macraig) [reply doc/code_navigation/index.md:89]: Please ignore, this is the wrong section

**Review by fkling — APPROVED** [review]

**Pull request merged**